core/arch/arm/mm/tee_mmu.c

```diff
@@ -263,7 +263,7 @@ TEE_Result tee_mmu_init(struct tee_ta_ctx *ctx)
 TEE_Result tee_mmu_map(struct tee_ta_ctx *ctx, struct tee_ta_param *param)
 {
 	TEE_Result res = TEE_SUCCESS;
-	paddr_t pa;
+	paddr_t pa = NULL;
```

Comment:
You're assuming flat mapping now I guess. If we don't have flat mapping then physical address 0 can be represented by a virtual address which isn't 0.